passkey-client/src/lib.rs

```diff
@@ -367,6 +364,47 @@ where
             client_extension_results: AuthenticatorExtensionsClientOutputs::default(),
         })
     }
+
+    fn map_rk(&self, criteria: &Option<AuthenticatorSelectionCriteria>) -> bool {
+        match criteria.as_ref().unwrap_or(&Default::default()) {
+            // > If pkOptions.authenticatorSelection.residentKey:
+            // > is present and set to required
+            AuthenticatorSelectionCriteria {
+                resident_key: Some(ResidentKeyRequirement::Required),
+                ..
+            // > Let requireResidentKey be true.
+            } => true,
+
+            // > is present and set to preferred
+            // >  And the authenticator is capable of client-side credential storage modality
+            AuthenticatorSelectionCriteria {
+                resident_key: Some(ResidentKeyRequirement::Preferred),
+                ..
+            // > Let requireResidentKey be true.
+            } => true,
+
+            // > is present and set to preferred
+            // >  And is not capable of client-side credential storage modality, or if the client cannot determine authenticator capability,
+            // >  Let requireResidentKey be false.
+            // `passkey-rs` requires the authenticator to be capable of client-side credential storage modality,
+            // so we do not need to check for this case.
```

Comment:
**Suggestion:** We can pass in the result of `authenticator.get_info` and look at `get_info::Response::options::rk` for this

Reply:
Something like this? :)